protocol/xmrmaker/checks.go

```diff
@@ -171,3 +172,34 @@ func (s *swapState) checkContract(txHash ethcommon.Hash) error {
 
 	return nil
 }
+
+// checkAndSetTimeouts checks that the timeouts set by the counterparty when initiating the swap
+// are not too short or too long.
+// we expect the timeout to be of a certain length (1 hour for mainnet/stagenet), and allow a 3 minute
+// variation between now and the expected time until the first timeout t0, to allow for block confirmations.
+// the time between t0 and t1 should always be the exact length we expect.
+func (s *swapState) checkAndSetTimeouts(t0, t1 *big.Int) error {
+	s.setTimeouts(t0, t1)
+
+	expectedTimeout := common.SwapTimeoutFromEnvironment(s.Backend.Env())
+	allowableTimeDiff := expectedTimeout / 20
+
+	if s.Backend.Env() == common.Development {
+		allowableTimeDiff = time.Minute
+	}
+
+	if time.Now().Add(expectedTimeout).Sub(s.t0).Abs() > allowableTimeDiff {
+		return errInvalidT0
+	}
+
+	if s.t1.Sub(s.t0).Abs() == expectedTimeout {
```

Comment:
Was that supposed to be `!=`?

Reply:
If the time deltas don't turn out to be exact enough for an exact comparison, we can subtract the big Int values instead.

Reply:
yes, fixed